tool/util/fs.dart

```diff
@@ -11,30 +11,51 @@ pushd(String path) async {
   if (!(await dir.exists())) {
     await dir.create(recursive: true);
   }
-  Directory.current = dir;
+  cd(dir.path);
 }
 
-Future<dynamic> readJsonFile(String path, [defaultValue]) async {
+Future<dynamic> readJsonFile(String path, {defaultValue, String shadow}) async {
```

Comment:
Adds the JSON shadow concept, which is a second file that is merged with the file attempted to be read. That will make it so that you can override committed files locally. I use this to override what branch and repository to build from when running the builder locally, if for example I need to test a fix in production.